File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3VehicleDescriptor.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3VehicleDescriptor {
    /// <summary>
    /// Operator name of the vehicle such as \"Metro Trains Melbourne\", \"Yarra Trams\", \"Ventura Bus Line\", \"CDC\" or \"Sita Bus Lines\" . May be null/empty.  Only available for train, tram, v/line and some bus runs.
    /// </summary>
    /// <value>Operator name of the vehicle such as \"Metro Trains Melbourne\", \"Yarra Trams\", \"Ventura Bus Line\", \"CDC\" or \"Sita Bus Lines\" . May be null/empty.  Only available for train, tram, v/line and some bus runs.</value>
    [DataMember(Name="operator", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "operator")]
    public string _Operator { get; set; }

    /// <summary>
    /// Operator identifier of the vehicle such as \"26094\". May be null/empty. Only available for some tram and bus runs.
    /// </summary>
    /// <value>Operator identifier of the vehicle such as \"26094\". May be null/empty. Only available for some tram and bus runs.</value>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public string Id { get; set; }

    /// <summary>
    /// Indicator if vehicle has a low floor. May be null. Only available for some tram runs.
    /// </summary>
    /// <value>Indicator if vehicle has a low floor. May be null. Only available for some tram runs.</value>
    [DataMember(Name="low_floor", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "low_floor")]
    public bool? LowFloor { get; set; }

    /// <summary>
    /// Indicator if vehicle is air conditioned. May be null. Only available for some tram runs.
    /// </summary>
    /// <value>Indicator if vehicle is air conditioned. May be null. Only available for some tram runs.</value>
    [DataMember(Name="air_conditioned", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "air_conditioned")]
    public bool? AirConditioned { get; set; }

    /// <summary>
    /// Vehicle description such as \"6 Car Comeng\", \"6 Car Xtrapolis\", \"3 Car Comeng\", \"6 Car Siemens\", \"3 Car Siemens\". May be null/empty.  Only available for some metropolitan train runs.
    /// </summary>
    /// <value>Vehicle description such as \"6 Car Comeng\", \"6 Car Xtrapolis\", \"3 Car Comeng\", \"6 Car Siemens\", \"3 Car Siemens\". May be null/empty.  Only available for some metropolitan train runs.</value>
    [DataMember(Name="description", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "description")]
    public string Description { get; set; }

    /// <summary>
    /// Supplier of vehicle descriptor data.
    /// </summary>
    /// <value>Supplier of vehicle descriptor data.</value>
    [DataMember(Name="supplier", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "supplier")]
    public string Supplier { get; set; }

    /// <summary>
    /// The length of the vehicle. Applies to CIS - Metro Trains
    /// </summary>
    /// <value>The length of the vehicle. Applies to CIS - Metro Trains</value>
    [DataMember(Name="length", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "length")]
    public string Length { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3VehicleDescriptor {\n");
      sb.Append("  _Operator: ").Append(_Operator).Append("\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  LowFloor: ").Append(LowFloor).Append("\n");
      sb.Append("  AirConditioned: ").Append(AirConditioned).Append("\n");
      sb.Append("  Description: ").Append(Description).Append("\n");
      sb.Append("  Supplier: ").Append(Supplier).Append("\n");
      sb.Append("  Length: ").Append(Length).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: java-vertx/src/main/java/io/swagger/server/api/model/V3VehicleDescriptor.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3VehicleDescriptor
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3VehicleDescriptor   {
  private String operator = null;

  private String id = null;

  private Boolean lowFloor = null;

  private Boolean airConditioned = null;

  private String description = null;

  private String supplier = null;

  private String length = null;

  public V3VehicleDescriptor operator(String operator) {
    this.operator = operator;
    return this;
  }

  /**
   * Operator name of the vehicle such as \"Metro Trains Melbourne\", \"Yarra Trams\", \"Ventura Bus Line\", \"CDC\" or \"Sita Bus Lines\" . May be null/empty.  Only available for train, tram, v/line and some bus runs.
   * @return operator
   **/
    public String getOperator() {
    return operator;
  }

  public void setOperator(String operator) {
    this.operator = operator;
  }

  public V3VehicleDescriptor id(String id) {
    this.id = id;
    return this;
  }

  /**
   * Operator identifier of the vehicle such as \"26094\". May be null/empty. Only available for some tram and bus runs.
   * @return id
   **/
    public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public V3VehicleDescriptor lowFloor(Boolean lowFloor) {
    this.lowFloor = lowFloor;
    return this;
  }

  /**
   * Indicator if vehicle has a low floor. May be null. Only available for some tram runs.
   * @return lowFloor
   **/
    public Boolean isLowFloor() {
    return lowFloor;
  }

  public void setLowFloor(Boolean lowFloor) {
    this.lowFloor = lowFloor;
  }

  public V3VehicleDescriptor airConditioned(Boolean airConditioned) {
    this.airConditioned = airConditioned;
    return this;
  }

  /**
   * Indicator if vehicle is air conditioned. May be null. Only available for some tram runs.
   * @return airConditioned
   **/
    public Boolean isAirConditioned() {
    return airConditioned;
  }

  public void setAirConditioned(Boolean airConditioned) {
    this.airConditioned = airConditioned;
  }

  public V3VehicleDescriptor description(String description) {
    this.description = description;
    return this;
  }

  /**
   * Vehicle description such as \"6 Car Comeng\", \"6 Car Xtrapolis\", \"3 Car Comeng\", \"6 Car Siemens\", \"3 Car Siemens\". May be null/empty.  Only available for some metropolitan train runs.
   * @return description
   **/
    public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public V3VehicleDescriptor supplier(String supplier) {
    this.supplier = supplier;
    return this;
  }

  /**
   * Supplier of vehicle descriptor data.
   * @return supplier
   **/
    public String getSupplier() {
    return supplier;
  }

  public void setSupplier(String supplier) {
    this.supplier = supplier;
  }

  public V3VehicleDescriptor length(String length) {
    this.length = length;
    return this;
  }

  /**
   * The length of the vehicle. Applies to CIS - Metro Trains
   * @return length
   **/
    public String getLength() {
    return length;
  }

  public void setLength(String length) {
    this.length = length;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3VehicleDescriptor v3VehicleDescriptor = (V3VehicleDescriptor) o;
    return Objects.equals(this.operator, v3VehicleDescriptor.operator) &&
        Objects.equals(this.id, v3VehicleDescriptor.id) &&
        Objects.equals(this.lowFloor, v3VehicleDescriptor.lowFloor) &&
        Objects.equals(this.airConditioned, v3VehicleDescriptor.airConditioned) &&
        Objects.equals(this.description, v3VehicleDescriptor.description) &&
        Objects.equals(this.supplier, v3VehicleDescriptor.supplier) &&
        Objects.equals(this.length, v3VehicleDescriptor.length);
  }

  @Override
  public int hashCode() {
    return Objects.hash(operator, id, lowFloor, airConditioned, description, supplier, length);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3VehicleDescriptor {\n");
    
    sb.append("    operator: ").append(toIndentedString(operator)).append("\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    lowFloor: ").append(toIndentedString(lowFloor)).append("\n");
    sb.append("    airConditioned: ").append(toIndentedString(airConditioned)).append("\n");
    sb.append("    description: ").append(toIndentedString(description)).append("\n");
    sb.append("    supplier: ").append(toIndentedString(supplier)).append("\n");
    sb.append("    length: ").append(toIndentedString(length)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
